### PR TITLE
build: Fix windows build when the source path includes spaces

### PR DESCRIPTION
**Pull request by jeremyg-lunarg**

### CODE REVIEW

File: tests/layers/CMakeLists.txt [hunk 1]
```diff
@@ -23,7 +23,7 @@ add_library(VkLayer_device_profile_api MODULE
 target_link_libraries(VkLayer_device_profile_api PRIVATE VkLayer_utils)
 
 if (WIN32)
-    set_target_properties(VkLayer_device_profile_api PROPERTIES LINK_FLAGS "/DEF:${CMAKE_CURRENT_SOURCE_DIR}/VkLayer_device_profile_api.def")
+    set_target_properties(VkLayer_device_profile_api PROPERTIES LINK_FLAGS "/DEF:\"${CMAKE_CURRENT_SOURCE_DIR}/VkLayer_device_profile_api.def\"")
```
Comment (juan-lunarg) [inline tests/layers/CMakeLists.txt:26]: Out of curiosity is this still a problem if you use `target_link_options`?

Comment (jeremyg-lunarg) [reply tests/layers/CMakeLists.txt:26]: It looks like that works, I'll push again.

File: layers/CMakeLists.txt [hunk 2]
```diff
@@ -239,7 +239,7 @@ if (VVL_ENABLE_ASAN)
 endif()
 
 if(WIN32)
-    set_target_properties(VkLayer_khronos_validation PROPERTIES LINK_FLAGS "/DEF:${CMAKE_CURRENT_SOURCE_DIR}/VkLayer_khronos_validation.def")
+    target_link_options(VkLayer_khronos_validation INTERFACE "/DEF:${CMAKE_CURRENT_SOURCE_DIR}/VkLayer_khronos_validation.def")
```
Comment (jeremyg-lunarg) [inline layers/CMakeLists.txt:242]: I am just guessing with INTERFACE, should it be PUBLIC or PRIVATE instead?

Comment (juan-lunarg) [reply layers/CMakeLists.txt:242]: It should be `PRIVATE`. Note, we can't use `target_link_options` yet due to our cmake minimum being `3.10`.

**Review by juan-lunarg — APPROVED** [review]

**Pull request merged**